project: overhaul the logging

ffromani opened the pull request:
switch to `go-logger` over the `log` package from the stdlib.
The package was selected using these criterias (presented unordered), first fit:

1. compatible license (obviously)
2. no extra options added silently (looking at you, glog)
3. support for leveled logging of any form
4. avoid global context if possible

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
project: overhaul the logging

switch to `go-logger` over the `log` package from the stdlib.
The package was selected using these criterias (presented unordered), first fit:

1. compatible license (obviously)
2. no extra options added silently (looking at you, glog)
3. support for leveled logging of any form
4. avoid global context if possible

Signed-off-by: Francesco Romani <fromani@redhat.com>

fromanirh commented:
Fixes https://github.com/fromanirh/pack8s/issues/37

fromanirh commented:
/cc @ksimon1

ksimon1 commented on cmd/pull.go:
```diff
@@ -38,16 +40,14 @@ func NewPullCommand() *cobra.Command {
-		return err
-	}
+
+	hnd, err := podman.NewHandle(ctx, cOpts.PodmanSocket, cmdutil.NewLogger(cOpts.Verbose, 0))
```
I think you removed error handling by mistake

ffromani replied:
Indeed. Thanks.

ksimon1 commented on cmd/rm.go:
```diff
@@ -55,54 +56,50 @@ func (cl containerList) Swap(i, j int) {
-		return err
-	}
+	log := cmdutil.NewLogger(cOpts.Verbose, 0)
+	hnd, err := podman.NewHandle(ctx, cOpts.PodmanSocket, log)
```
same here

ffromani committed:
cmd: restore missing error handling

it was killed by a bad rebase, we need it back

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
log: disable colors by default

pack8s is meant to be used non-interactively, so the
colors in the logging messages are most often more harmful
than useful.

Hence, we remove the tunable from the internal logging helper.
We still allow to enable them with a new environment variable,
PACK8S_COLORS

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
scp: improve the log message

make it clearer what's being scopied where

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
options: fix help message

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
ignore .kube directory

it is created when running functests

Signed-off-by: Francesco Romani <fromani@redhat.com>

fromanirh commented:
I'll cut another release once this PR is merged.

ksimon1 reviewed (approved):
LGTM
